[connector/servicegraph] Update virtual node defaults and change feature gate to beta

ogxd opened the pull request:
**Description:**

- Change default values to peer.service, db.name and db.system (in this order). These are compliant with conventions, and the same as the tempo service graph uses.
- Switch the feature from "alpha" to "beta". Virtual nodes are working well on the tempo side, and feature could be disabled anyway by using an empty list if needed.
- Reword documentation for improved clarity
- Fix "processor" that should be "connector" instead since recent commit

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31734

**Testing:** No functional change requiring additional tests

**Documentation:** Updated documentation

ogxd committed:
[connector/servicegraph] Update virtual node defaults and change feature gate to beta

ogxd committed:
[connector/servicegraph] Update virtual node defaults and change feature gate to beta

ogxd commented on connector/servicegraphconnector/README.md:
```diff
@@ -120,7 +120,7 @@ datasources:
 The following settings are required:
 
-- `metrics_exporter`: the name of the exporter that this processor will write metrics to. This exporter **must** be present in a pipeline.
+- `metrics_exporter`: the name of the exporter that this connector will write metrics to. This exporter **must** be present in a pipeline.
```
Is this still relevant since this is now a connector? I managed to get the connector working without using this field.

ogxd committed:
Update feature gate description

ogxd committed:
Update feature gate description

mapno reviewed (approved):
This a breaking change, but since the feature was in alpha I think it's still ok to change the attribute list. It now matches the Tempo defaults, so I'm good with the change. Thanks!

ogxd commented:
I have updated the defaults in [the feature gate description](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31735/commits/201df8d4cad202c6e0046106365234cd3abeb505) as well, I'll need another approval 🙂

ogxd committed:
Merge branch 'main' into servicegraphconnector-virtualnodes

ogxd commented:
@jpkrohling can this be merged?

JaredTan95 reviewed (approved):
LGTM

ogxd commented:
@jpkrohling can this be merged?

Frapschen commented:
@ogxd Sorry for the delay, I think we should add a changelog to notify users

Frapschen commented on connector/servicegraphconnector/factory.go:
```diff
@@ -30,8 +30,8 @@ var virtualNodeFeatureGate, legacyMetricNamesFeatureGate, legacyLatencyUnitMsFea
-		featuregate.StageAlpha,
-		featuregate.WithRegisterDescription("When enabled, when the edge expires, processor checks if it has peer attributes(`db.name, net.sock.peer.addr, net.peer.name, rpc.service, http.url, http.target`), and then aggregate the metrics with virtual node."),
+		featuregate.StageBeta,
+		featuregate.WithRegisterDescription("When enabled, when the edge expires, processor checks if it has peer attributes(`peer.service, db.name, db.system`), and then aggregate the metrics with virtual node."),
```
```suggestion
		featuregate.WithRegisterDescription("Enabled by default, when the edge expires, processor checks if it has peer attributes(`peer.service, db.name, db.system`), and then aggregate the metrics with virtual node."),
```

ogxd replied:
Other beta features also start with "When enable" so I wanted to avoid breaking uniformity, but this description could indeed be improved, here is what I came up with:

> When enabled and setting `virtual_node_peer_attributes` is not empty, the connector looks for the presence of these attributes in span to create virtual server nodes.

Frapschen commented on connector/servicegraphconnector/README.md:
```diff
@@ -136,8 +136,8 @@ The following settings can be optionally configured:
   - Default: `2s`
-- `virtual_node_peer_attributes`: the list of attributes need to match for building virtual server node, the higher the front, the higher the priority.
-  - Default: `[db.name, net.sock.peer.addr, net.peer.name, rpc.service, net.sock.peer.name, net.peer.name, http.url, http.target]`
+- `virtual_node_peer_attributes`: the list of attributes, ordered by priority, whose presence in a client span will result in the creation of a virtual server node.
```
Since we will enable it by default, we also need to document how to disable it, it's a high-cardinality option.

Frapschen commented:
> feature could be disabled anyway by using an empty list if needed.

I doubt that not work. 

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/626046f70f5c8528d74ede0cf4808e04491c3d3c/connector/servicegraphconnector/connector.go#L383-L394

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/626046f70f5c8528d74ede0cf4808e04491c3d3c/connector/servicegraphconnector/connector.go#L633-L642

Even when the `VirtualNodePeerAttributes` is empty, a virtual node service named `unknown` will exist when this feature gate is enabled by default.

ogxd committed:
Merge branch 'servicegraphconnector-virtualnodes' of https://github.com/ogxd/opentelemetry-collector-contrib into servicegraphconnector-virtualnodes

ogxd committed:
Add a way to disable virtual server nodes feature

ogxd committed:
Update feature description

ogxd committed:
Merge branch 'main' into servicegraphconnector-virtualnodes

ogxd committed:
Add changelog entry

ogxd commented:
Indeed, I just added a check to disable the feature if `virtual_node_peer_attributes` is empty. I also updated the documentation and included a changelog entry.

ogxd committed:
Merge branch 'main' into servicegraphconnector-virtualnodes

ogxd committed:
Merge branch 'main' into servicegraphconnector-virtualnodes

Frapschen reviewed (approved):
LGTM

ogxd committed:
Merge branch 'main' into servicegraphconnector-virtualnodes

ogxd commented:
Please excuse me for the bothering 😅
Can this PR be merged? @Frapschen @jpkrohling

ogxd committed:
Merge branch 'main' into servicegraphconnector-virtualnodes

ogxd committed:
Merge branch 'main' into servicegraphconnector-virtualnodes
